site/app/libraries/Core.php

```diff
@@ -118,7 +118,7 @@ public function disableRedirects() {
      * @param $course
      * @throws \Exception
      */
-    public function loadConfig($semester, $course) {
+    public function loadCourseConfig($semester, $course) {
         $conf_path = FileUtils::joinPaths(__DIR__, '..', '..', '..', 'config');
 
         $this->config = new Config($this, $semester, $course);
```

Comment:
modify this such that `if ($this->config === null) { throw Exception("Master config has not been loaded") }` at the top and then change `Config::loadCourseJson($course_json_path)` to be `Config::loadCourseJson($semester, $course, $course_json_path)` and set the class properties for those two values in that function.